### DISCUSSION TITLE
Is `#[ast_serde]` still useful?

### DISCUSSION THREAD

**Discussion by hyf0** [swc-project/swc#5303] [General]
I'm trying to be familiar with the codebase. Is this comment still meaningful?

https://github.com/swc-project/swc/blob/2017daa2d5078ad80fb88cbddfd6fa5a80c13eba/crates/ast_node/src/lib.rs#L161

**Comment by kdy1** [comment]
Oh it's not delete but it should be `delegate`

**Reply by hyf0** [reply]
If I understand correctly, It means that `#[ast_node]` will covers the usage of `#[ast_serde]`, and all the places used `#[ast_serde]` will e replaced with `#[ast_node]`?

**Reply by kdy1** [reply]
No, it means #[ast_node] will call #[ast_serde] by generating attrs

**Reply by hyf0** [reply]
Is this a good first issue? If it is, I will give it a try.

**Reply by kdy1** [reply]
I don't think so as it will make compile time longer.
Calling it directly from proc macro is better

**Reply by hyf0** [reply]
Well, I guess I could delete this comment at least? :)

**Reply by kdy1** [reply]
Yrah I think so, thanks!
